src/beast/beast/utility/impl/PropertyStream.cpp

```diff
@@ -192,37 +193,31 @@ std::string const& PropertyStream::Source::name () const
 
 void PropertyStream::Source::add (Source& source)
 {
-    SharedState::Access state (m_state);
-    SharedState::Access childState (source.m_state);
-    bassert (childState->parent == nullptr);
-    state->children.push_back (childState->item);
-    childState->parent = this;
+    std::lock_guard<std::recursive_mutex> _l(lock_);
+    std::lock_guard<std::recursive_mutex> _cl(source.lock_);
```

Comment:
Our ordering is well defined: grab parent, grab child. This should never fail us, since we're dealing with a tree structure (i.e. an acyclic graph) which means an item's parents can't be its child at the same time.

However, using `std::lock` seems like a good idea anyways. Fixed.